adaptivemd/engine/openmm/openmmrun.py

```diff
@@ -33,6 +33,88 @@
 from simtk.openmm import Platform, XmlSerializer
 from simtk.openmm.app import PDBFile, Simulation, DCDReporter, StateDataReporter
 
+
+
+def get_xml(xml_file):
+    # TODO file access control
+    attempt = 0
+    retries = 500
+    while True:
+        try:
+            with open(xml_file) as f:
+                xml = f.read()
+                cereal = XmlSerializer.deserialize(xml)
+            return xml, cereal
+
+        except ValueError as e:
+            if attempt < retries:
+                attempt += 1
+                time.sleep(5*random.random())
+            else:
+                raise e
+
+
+def get_platform(platform_name):
+    if platform_name == 'fastest':
+        platform = None
+    else:
+        # TODO file access control
+        attempt = 0
+        retries = 500
+        while True:
+            try:
+                platform = Platform.getPlatformByName(platform_name)
+                return platform
+
+            except IndexErrorError as e:
+                if attempt < retries:
+                    attempt += 1
+                    time.sleep(5*random.random())
+                else:
+                    raise e
+
+
+def get_pdbfile(topology_pdb):
+    # TODO file access control
+    attempt = 0
+    retries = 500
+    while True:
+        try:
+            pdb = PDBFile(topology_pdb)
+            return pdb
+
+        except IndexError as e:
+            if attempt < retries:
+                attempt += 1
+                time.sleep(5*random.random())
+            else:
+                raise e
+
+def read_input(platform, pdbfile, system, integrator):
+    import time
+    import random
+
+    return_order = ['get_platform', 'get_pdbfile',
+                    'get_system', 'get_integrator']
+
+    funcs = {'get_platform':   [get_platform, platform],
+             'get_pdbfile':    [get_pdbfile, pdbfile],
+             'get_system':     [get_xml, system],
+             'get_integrator': [get_xml, integrator]}
+
+    kfuncs = list(funcs.keys())
+    random.shuffle(kfuncs)
```

Comment:
Why??

Reply:
When running a large number of workers who start tasks simultaneously (100s to 1000s), there will be filesystem errors. On the Titan lustre filesystem, the rate of errors was maybe 0.1 to 0.5%. This function shuffles the file read order, desychronizing to help alleviate this a bit.